connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/DataTypeProtoConverter.scala

```diff
@@ -79,6 +79,9 @@ object DataTypeProtoConverter {
     }
   }
 
+  private def toCatalystStringType(t: proto.DataType.String): StringType =
+    StringType(t.getCollationId)
```

Comment:
`t.getCollationId` can be null if the message was send from old-version clients. Shall we be more lenient here and use the default collation if it's null?

Reply:
nvm, I just saw https://github.com/apache/spark/pull/45233#discussion_r1504190868